script/signup.js

```diff
@@ -0,0 +1,123 @@
+import {
+  createErrorMessage,
+  authenticateUser,
+  isEmailValid,
+  displayError,
+  clearError,
+  handlePasswordInputFocusOut,
+  handlePasswordCheckInputFocusOut,
+  handleEmailInputFocusIn,
+  togglePasswordVisibility,
+} from './utils.js';
+
+import { VALID_USER } from './constants.js';
+
+document.addEventListener('DOMContentLoaded', function () {
+  const emailInput = document.getElementById('email');
+  const passwordInput = document.getElementById('password');
+  const passwordcheckInput = document.getElementById('passwordcheck');
+  const signInButton = document.getElementById('cta');
+  const watchPassword = document.getElementById('eye-button');
+
+  // 에러 메시지 요소들 생성 및 초기화
+  const emailErrorMessage = createErrorMessage('email-error', '');
+  const passwordErrorMessage = createErrorMessage('password-error', '');
+  const passwordCheckErrorMessage = createErrorMessage('password-check-error', ''); // 추가: 비밀번호 확인 에러 메시지
+
+  // 이메일, 비밀번호 입력란 뒤에 에러 메시지 요소 추가
+  emailInput.after(emailErrorMessage);
+  passwordInput.after(passwordErrorMessage);
+  passwordcheckInput.after(passwordCheckErrorMessage); // 추가: 비밀번호 확인 에러 메시지
+
+  // 이메일 입력란 focusin 이벤트 핸들러
+  emailInput.addEventListener('focusin', function () {
+    clearError(emailInput, emailErrorMessage);
+  });
+
+  // 이메일 입력란 focusout 이벤트 핸들러
+  emailInput.addEventListener('focusout', function () {
+    handleEmailInputFocusIn(emailInput, emailErrorMessage);
+    // 추가: 이미 사용 중인 이메일 여부 검사
+    if (emailInput.value.trim() === VALID_USER.email) {
+      displayError(emailInput, emailErrorMessage, '이미 사용 중인 이메일입니다.');
+    }
+  });
+
+  // 비밀번호 입력란 focusout 이벤트 핸들러
+  passwordInput.addEventListener('focusout', function () {
+    handlePasswordInputFocusOut(passwordInput, passwordErrorMessage);
+  });
+
+  // 비밀번호 확인 입력란 focusout 이벤트 핸들러
+  passwordcheckInput.addEventListener('focusout', function () {
+    handlePasswordCheckInputFocusOut(passwordcheckInput, passwordCheckErrorMessage, passwordInput);
+  });
+
+  // 비밀번호 확인 입력란 focusout 이벤트 핸들러에서 비밀번호 일치 여부 체크
+  function handlePasswordCheckInputFocusOut(
+    passwordcheckInput,
+    passwordCheckErrorMessage,
+    passwordInput
+  ) {
+    const passwordCheckValue = passwordcheckInput.value.trim();
+    const passwordValue = passwordInput.value.trim();
+
+    if (!passwordCheckValue) {
+      displayError(passwordcheckInput, passwordCheckErrorMessage, '비밀번호를 다시 입력해주세요.');
+    } else if (passwordCheckValue !== passwordValue) {
+      displayError(passwordcheckInput, passwordCheckErrorMessage, '비밀번호가 일치하지 않아요.');
+    } else {
+      clearError(passwordcheckInput, passwordCheckErrorMessage);
+    }

```

Comment:
if-elseif-else 를 얼리 리턴을 사용해서 리팩터링해볼까요?
elseif-else 를 사용하면 좋은 부분도 있겠지만, 대체로는 불필요하게 가독성을 해치거든요 ㅎㅎ

```suggestion
    if (!passwordCheckValue) {
      displayError(passwordcheckInput, passwordCheckErrorMessage, '비밀번호를 다시 입력해주세요.');
      return
    } 
    if (passwordCheckValue !== passwordValue) {
      displayError(passwordcheckInput, passwordCheckErrorMessage, '비밀번호가 일치하지 않아요.');
      return;
    }
      clearError(passwordcheckInput, passwordCheckErrorMessage);
    
```

Reply:
[1596314](https://github.com/codeit-bootcamp-frontend/4-Weekly-Mission/pull/390/commits/1596314bf41a44054db13a4286ff2ac294ed18cb) 위 커밋과 같지만 얼리 리턴 적용 완료했습니다!